File: app/src/main/java/de/datlag/hotdrop/auth/InteraAccount.java
package de.datlag.hotdrop.auth;

import com.google.gson.Gson;
import com.google.gson.JsonObject;

import lombok.Getter;
import lombok.Setter;

public class InteraAccount {

    @Getter
    private String response;

    private JsonObject object;

    @Getter
    @Setter
    private String username;

    @Getter
    @Setter
    private boolean error = true;

    @Getter
    @Setter
    private String email;

    @Getter
    @Setter
    private String description;

    @Getter
    @Setter
    private boolean verified = false;

    public InteraAccount(String response) {
        this.response = response;
        init();
    }

    private void init() {
        object = new Gson().fromJson(response, JsonObject.class);
        setError(object.get("error").getAsBoolean());

        if (!error) {
            setUsername(object.get("username").getAsString());
            setEmail(object.get("mail").getAsString());
            setDescription(object.get("description").getAsString());
            setVerified(object.get("verified").getAsBoolean());
        }
    }

}

File: app/src/main/java/de/datlag/hotdrop/auth/InteraAccount.kt
package de.datlag.hotdrop.auth

import com.google.gson.Gson
import com.google.gson.JsonObject
import lombok.Getter
import lombok.Setter

class InteraAccount(response: String) {
    private var `object`: JsonObject = Gson().fromJson(response, JsonObject::class.java)
    private var error = true
    var username: String? = null
    var email: String? = null
    var description: String? = null
    var verified = false

    init {
        error = `object`.get("error").asBoolean
        if (!error) {
            username = `object`.get("username").asString
            email = `object`.get("username").asString
            description = `object`.get("description").asString
            verified = `object`.get("verified").asBoolean
        }
    }
}